runtime/src/main/cpp/Worker.cpp

```diff
@@ -726,12 +610,218 @@ KNativePtr detachObjectGraphInternal(KInt transferMode, KRef producer) {
 
 }  // namespace
 
+Worker* WorkerInit(KBoolean errorReporting) {
+#if WITH_WORKERS
+  if (::g_worker != nullptr) return ::g_worker;
+  Worker* worker = theState()->addWorkerUnlocked(errorReporting != 0);
+  ::g_worker = worker;
+  return worker;
+#else
+  return nullptr;
+#endif  // WITH_WORKERS
+}
+
+void WorkerDeinit(Worker* worker) {
+#if WITH_WORKERS
+  ::g_worker = nullptr;
+  theState()->destroyWorkerUnlocked(worker);
```

Comment:
Two different worker variables involved. Maybe add an assertion to check that the values are the same?

Reply:
They don't have to, `g_worker` could be zeroed out by TLS deinit already.